sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -367,9 +367,12 @@ message WriteFilesPayload {
   // (Required) The SdkFunctionSpec of the FileBasedSink.
   SdkFunctionSpec sink = 1;
 
-  bool windowed_writes = 2;
+  // (Required) The format function.
```

Comment:
Yes. This is an exact mirror of what was done BigQueryIO. DynamicDestinations makes sense to apply to the user type, not always to the type written to the file which might not contain the necessary information to pick a destination. The format function allows the user to supply a UDT to the sink, and convert to the sink's native type before writing. In general this will just be serialized inside WriteFiles.